code/modules/mob/living/carbon/human/examine.dm

```diff
@@ -381,6 +382,43 @@
 
 	to_chat(user, msg.Join())
 
+	if(ishuman(user))
+		var/mob/living/carbon/human/U = user
+		if(U.species.eye_contact_message && distance < EYECONTACTDIS && !U.stat)
+			var/eyescovered = FALSE
+			for(var/obj/item/clothing/C in list(U.wear_mask, U.head, U.glasses))
+				if(C.body_parts_covered & EYES)
+					eyescovered = TRUE
+					break
+			if(!eyescovered)
+				U.handle_examine(src)
+				LAZYADD(U.recent_examines, WEAKREF(src)) //Done after so that it doesn't work with itself
+				addtimer(CALLBACK(U, .proc/removeexamine, src), 5 SECONDS)
+
+// Helper proc for the timer above. 
+/mob/living/carbon/human/proc/removeexamine(item)
+	LAZYREMOVE(recent_examines, item)
+
+//Should be called on the mob that is doing the examining by examine() above. Checks the examined mob's recent examine list; if we are in that list, then
+//it is safe to say that they examined us recently, so we should make eye contact.
+//Returns TRUE if we successfully made eye contact, false otherwise.
+/mob/living/carbon/human/proc/handle_examine(mob/living/carbon/human/examinee)
+	if(examinee == src || !examinee.recent_examines || !LAZYLEN(examinee.recent_examines))
+		return FALSE
+
+	var/datum/weakref/WR = WEAKREF(src)
+	if(WR in examinee.recent_examines)
+		if(src.species != examinee.species)
+			to_chat(src, SPAN_SUBTLE(replacetext(examinee.species.eye_contact_message, "%SRCNAME%", "\the [examinee]")))
+			to_chat(examinee, SPAN_SUBTLE(replacetext(src.species.eye_contact_message, "%SRCNAME%", "\the [src]")))
+		else
+			to_chat(src, SPAN_SUBTLE("You make eye contact with \the [examinee]."))
+			to_chat(examinee, SPAN_SUBTLE("You make eye contact with \the [src]."))
+		recent_examines -= examinee   //Remove them from our list
```

Comment:
Sorry about the wait.

This line runtimes. `examinee` is a `mob/living/carbon/human`, but `recent_examines` is a list of weakrefs. I've not done much with weakrefs myself so I don't know the best course of action here, but you'll probably have to resolve each weakref to its actual mobtype and check it against examinee, and then remove any matches. 

Also, you're using lazy helpers with `recent_examines`, but you never `LAZYINITLIST(recent_examines)` anywhere. I'd probably stick that in some human init somewhere just to avoid weird list behavior. (I'm actually surprised it works without that and/or `recent_examines = list()` but the more you know)

Reply:
the `LAZYINITLIST(recent_examines)` solved the runtime it seems so it does need that to work fully. Still surprising it did almost work otherwise.

Reply:
putting LAZYINITLIST in human init makes the lazylist meaningless, since the whole point is to sacrifice processing to spare memory. if we have every human running around with a list, that's like 4kb of lists being stored, way too much /s

replacing the `recent_examines -= examinee` with `LAZYREMOVE(recent_examines, WEAKREF(examinee))` should be fine? i think